src/main/java/uk/gov/companieshouse/acsp/service/FilingsService.java

```diff
@@ -139,7 +147,7 @@ private void buildAcspData(Map<String, Object>data, AcspDataDto acspDataDto, Tra
       data.put("registered_office_address", buildBusinessAddress(acspDataDto));
     }
     if(acspDataDto.getBusinessAddress() != null &&
-            acspDataDto.getBusinessAddress().equals(acspDataDto.getCorrespondenceAddress())) {
+            acspDataDto.getBusinessAddress().equals(acspDataDto.getApplicantDetails().getCorrespondenceAddress())) {
```

Comment:
null check required here for applicantDetails

Reply:
added.